.github/copilot-instructions.md

```diff
@@ -0,0 +1,249 @@
+
+---
+
+## 5. Libraries Workflow
+
+From the repository root:
+
+- Build:
+  `./build.sh -subset libs -rc release`
+
+- Run tests:
+  `./build.sh -subset libs.tests -test -rc release`
+
+- More info:
+    - [Build Libraries](https://github.com/dotnet/runtime/blob/main/docs/workflow/building/libraries/README.md)
+    - [Testing Libraries](https://github.com/dotnet/runtime/blob/main/docs/workflow/testing/libraries/testing.md)
+
+### 5.1. How To: Identify Affected Libraries
+
+For each changed file under `src/libraries/`, find the matching library and its test project(s).
+Most libraries use:
+
+- Source: `src/libraries/<LibraryName>/src/<LibraryName>.csproj`
+
+- Tests (single):
+    - `src/libraries/<LibraryName>/tests/<LibraryName>.Tests.csproj`
+    - OR `src/libraries/<LibraryName>/tests/<LibraryName>.Tests/<LibraryName>.Tests.csproj`
+
+- Tests (multiple types):
+    - `src/libraries/<LibraryName>/tests/FunctionalTests/<LibraryName>.Functional.Tests.csproj`
+    - `src/libraries/<LibraryName>/tests/UnitTests/<LibraryName>.Unit.Tests.csproj`
+    - Or similar.
+
+---
+
+### 5.2. How To: Build and Test Specific Library
+
+If only one library is affected:
+
+1. **Navigate to the library directory:**
+   `cd src/libraries/<LibraryName>`
+
+2. **Build the library:**
+   `dotnet build`
+
+3. **Build and run all test projects:**
+
+    - For each discovered `*.Tests.csproj` in the `tests` subdirectory:
+      `dotnet build /t:test ./tests/<TestProject>.csproj`
+
+        - *Adjust path as needed. If in doubt, search with `find tests -name '*.csproj'`.*
+
+    - `dotnet build /t:test` is generally preferred over `dotnet test`
+
+---
+
+## 6. WebAssembly (WASM) Libraries Workflow
+
+From the repository root:
+
+- Build:
+  `./build.sh libs -os browser`
+
+- Run tests:
+  `./build.sh libs.tests -test -os browser`
+
+- More info:
+    - [Build libraries for WebAssembly](https://github.com/dotnet/runtime/blob/main/docs/workflow/building/libraries/webassembly-instructions.md)
+    - [Testing Libraries on WebAssembly](https://github.com/dotnet/runtime/blob/main/docs/workflow/testing/libraries/testing-wasm.md)
+
+---
+
+## 7. Additional Notes
+
+### 7.1. Troubleshooting
+
+- **Shared Framework Missing**
+
+    - If the build fails with an error "The shared framework must be built before the local targeting pack can be consumed.", build the runtime (clr or mono) first.
+      E.g., from the repo root, run `./build.sh clr -rc release`.
+
+- **Testhost Is Missing**
+
+    - If a test run fails with errors indicating a missing testhost, such as:
+        - "Failed to launch testhost with error: System.IO.FileNotFoundException", or
+        - "artifacts/bin/testhost/... No such file or directory",
+      this means the required runtime was not built.
+
+    - To resolve, build the appropriate runtime before running tests.
+      For example, run `./build.sh clr -rc release` if testing libraries.
```

Comment:
These don't look right to me.  Building the testhost is both clr and libs, in particular https://github.com/dotnet/runtime/blob/main/src/libraries/pretest.proj

Maybe we just remove this now that you're pre-building them?

Reply:
The case for this failure was when the agent used `./build.sh libs -rc release`, i.e. skipped `clr` from `clr+libs`. That's why I assumed that bringing the missing part should help. But if the `a+b` combination actually brings something additional over the two things `a` and `b` done separately, then yeah, it wouldn't work, but I must say it's a bit counter-intuitive 😅

I worry a bit that removing it completely will bring problems with a similar case but on a mono runtime, but if it does, we can address it separately.